### DISCUSSION TITLE
IPv6 / CoAP support ?

### DISCUSSION THREAD

**Discussion by tochinet** [JAndrassy/Arduino-Networking-API#10] [Ideas]
Hi, I find very interesting that you look at all the different network API implementations existing for Arduino, there are so many. But did you consider extending the scope to IPv6 as well ? In particular, while the ESP and W5x00 chips are clearly oriented towards IPv4, it is interesting to see that some others like the "good old' ENC28J60 may have a second life since its IP layer is in SW (AFAIK) and its internal memory can be a key into the support of IP for the 2k RAM constrained ATmega boards.

**Comment by JAndrassy** [comment]
I researched the API and from that perspective IPv6 doesn't make a difference. The API just uses the IPAddress class to get and set IP addresses of both types.

Esspressif supports IPv6 for ESP32 and esp8266 arduino community supports IPv6 too. You can use W5500 with both of them since they use it at `MAC level`. That of course applies to ENC28J60 too.

**Reply by tochinet** [reply]
I'm surprised by your answer. Just googling ESP32 IPv6 gets in first reply ([reddit last year](https://www.reddit.com/r/arduino/comments/16o9yj0/is_it_passible_to_connect_esp32_with_ipv6_internet/))

**_Yes, it is possible for just about any networked hardware to implement. There are implementations on ESP32 chips, STM32 chips, various Cortex designs, and more all have IPv6 implementations. 
No, it is not built in for Arduino. Much like your computer, there are drivers and libraries that need to communicate with different hardware. The default libraries and hardware drivers in Arduino do not have support for it._**

So I wouldn't just say it "doesn't make a difference".

**Reply by JAndrassy** [reply]
a year is a long time. a new version of esp32 arduino was released this year
